chromium_src/third_party/blink/renderer/modules/xr/xr.cc

```diff
@@ -0,0 +1,16 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/renderer/brave_content_settings_agent_impl_helper.h"
+
+#define BRAVE_XR_IS_SESSION_SUPPORTED        \
+  if (!AllowFingerprinting(frame)) {         \
```

Comment:
should be also be checking the content settings for the `session_mode` so we can return true if permission has already been granted?
```ContentSettingsType GetRequiredPermission(device::mojom::XRSessionMode mode) {
  switch (mode) {
    case device::mojom::XRSessionMode::kInline:
      return ContentSettingsType::SENSORS;
    case device::mojom::XRSessionMode::kImmersiveVr:
      return ContentSettingsType::VR;
    case device::mojom::XRSessionMode::kImmersiveAr:
      return ContentSettingsType::AR;
  }
}```

Reply:
cc @pes10k

Reply:
I don't think so because according to the spec, we're only blocking this at max farbling level, and at max farbling level it's an unconditional block.

Reply:
@bridiver I think an unconditional block makes sense here.  The general flow is to ask if the device is supported, and if so ask permission.  So with the above changes, there won't ever be a case where someone has given permission and is also in max protection mode (unless the person gave permission in default mode, and then jumped into max protection mode, in which case the cow is out of the barn and the person has given much stronger identification powers away)